src/Week12/Risk.hs

```diff
@@ -0,0 +1,90 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+module Week12.Risk where
+
+import Control.Monad.Random
+import Data.List
+
+------------------------------------------------------------
+-- Die values
+
+newtype DieValue = DV { unDV :: Int } 
+  deriving (Eq, Ord, Show, Num)
+
+first :: (a -> b) -> (a, c) -> (b, c)
+first f (a, c) = (f a, c)
+
+instance Random DieValue where
+  random           = first DV . randomR (1,6)
+  randomR (low,hi) = first DV . randomR (max 1 (unDV low), min 6 (unDV hi))
+
+die :: Rand StdGen DieValue
+die = getRandom
+
+------------------------------------------------------------
+-- Risk
+
+-- Given the definitions
+-- type Army = Int
+-- data Battlefield = Battlefield { attackers :: Army, defenders :: Army } 
+-- (which are also included in Risk.hs), write a function with the type
+-- battle :: Battlefield -> Rand StdGen Battlefield
+-- which simulates a single battle (as explained above) between two opposing armies. 
+-- That is, it should simulate randomly rolling the appropriate number of dice, interpreting the results, and updating the two armies to reflect casualties. 
+-- You may assume that each player will attack or defend with the maximum number of units they are allowed.
+
+type Army = Int
+
+data Battlefield = Battlefield { attackers :: Army, defenders :: Army }
+
+dice :: Int -> Rand StdGen [DieValue]
+dice n = sequence $ replicate n die
```

Comment:
Nitpick: you can use [replicateM](https://hackage.haskell.org/package/base-4.14.0.0/docs/Control-Monad.html#v:replicateM) to make this even simpler